pkg/rslog/output_builder_test.go

```diff
@@ -216,6 +299,56 @@ func (s *LoggerSuite) TestLogOutputTypeUnmarshalText() {
 	}
 }
 
+func (s *LoggerSuite) TestMultipleOutput() {
+	mockFileSystem := &MockFileSystem{}
+
+	builder := outputBuilder{
+		fileSystem:  mockFileSystem,
+		defaultFile: "/default/log/logfile.log",
+	}
+
+	fileExpectedWriter := &ioWriterMock{}
+	stdoutExpectedWriter := &ioWriterMock{}
+	stderrExpectedWriter := &ioWriterMock{}
+
+	mockFileSystem.On("Stat", "/custom").Return(mockFileInfo{}, nil)
+	mockFileSystem.On(
+		"OpenFile",
+		"/custom/test.log",
+		os.O_APPEND|os.O_CREATE|os.O_WRONLY,
+		os.FileMode(0600),
+	).Return(fileExpectedWriter, nil)
+	mockFileSystem.On("Stderr").Return(stderrExpectedWriter)
+	mockFileSystem.On("Stdout").Return(stdoutExpectedWriter)
+
+	outputDest := []OutputDest{
+		{
+			Output:   LogOutputFile,
+			Filepath: "/custom/test.log",
+		},
+		{
+			Output: LogOutputStdout,
+		},
+		{
+			Output: LogOutputStderr,
+		},
+	}
+
+	result, err := builder.Build(outputDest...)
+	s.Nil(err)
+
+	content := []byte("mock-content")
+	fileExpectedWriter.On("Write", content).Return(len(content), nil)
```

Comment:
The `On` calls prepare the mock object to receive the function calls and know how it will behave for that situation.
In the `AssertExpectations` function, try to ensure that every expectation you've prepared was triggered. 

So, If you want to have your mock object answering a function call, you will only use the `On` method.
If your test phase doesn't reach the mock call you prepared, there is no problem; it won't take the test to fail.

If you want to fail if the mock call was not triggered, you need to use the `AssertExpectations`.